config/routes.rb

```diff
@@ -1,8 +1,12 @@
 Myflix::Application.routes.draw do
   resources :categories
 
-  get 'videos', to: 'videos#index', as: "videos_path"
-  get 'videos/:id',to: 'videos#show', as: "video"
+  resources :videos, only: [:index,:show] do
+    collection do
+      post :search
```

Comment:
I would recommend using `get` for searches. That way the query shows in the address bar and is bookmarkable.